src/middlewares/index.ts

```diff
@@ -0,0 +1,60 @@
+import { Middleware } from './types'
+import { ActionTypes as PendingOrdersActionTypes } from 'reducers-actions/pendingOrders'
+
+import { setStorageItem } from 'utils'
+import { GP_PENDING_ORDER_KEY } from 'const'
+import { GlobalState } from 'reducers-actions'
+
+const logger: Middleware = getState => next => (action): void => {
+  process.env.NODE_ENV === 'development' &&
+    console.debug(`
+        =====> 
+        ACTION FIRED: ${action.type}
+        --
+        ACTION PAYLOAD: ${action.payload}
+        --
+        PREVIOUS STATE: ${getState()}
+        =====>
+    `)
+
+  // run next middleware and/or dispatch final
+  return next(action)
+}
+
+const pendingOrdersAndLocalStorage: Middleware = getState => next => (action): void => {
+  if (
+    action.type === PendingOrdersActionTypes.SAVE_PENDING_ORDERS ||
+    action.type === PendingOrdersActionTypes.REMOVE_PENDING_ORDERS
+  ) {
+    const { pendingOrders: state } = getState()
+
+    process.env.NODE_ENV === 'development' &&
+      console.debug(`
+        =====> 
+        PENDING ORDERS UPDATED
+    
+        P-ORDERS:
+        ${JSON.stringify(action.payload.orders, undefined, 2)}
+        =====>
+    `)
+
+    const { networkId, orders, userAddress } = action.payload
+
+    let newState: GlobalState['pendingOrders']
```

Comment:
Wouldn't be simpler to just add here `PendingOrdersState` as the type?

Reply:
yep